packages/plugins/graphql/server/src/format-graphql-error.ts

```diff
@@ -14,31 +9,64 @@ const formatErrorToExtension = (error: any) => ({
   error: pick(['name', 'message', 'details'])(error),
 });
 
-export function formatGraphqlError(error: GraphQLError) {
-  const { originalError } = error;
+function createFormattedError(
+  formattedError: GraphQLFormattedError,
+  message: string,
+  code: string,
+  originalError: unknown
+) {
+  const options = {
+    ...formattedError,
+    extensions: {
+      ...formattedError.extensions,
+      ...formatErrorToExtension(originalError),
+      code,
+    },
+  };
 
+  return new GraphQLError(message, options);
+}
+
+/**
+ * The handler for Apollo Server v4's formatError config option
+ *
+ * Intercepts specific Strapi error types to send custom error response codes in the GraphQL response
+ */
+export function formatGraphqlError(formattedError: GraphQLFormattedError, originalError: unknown) {
+  // If this error doesn't have an associated originalError, it
   if (isEmpty(originalError)) {
-    return error;
+    return formattedError;
   }
 
+  const { message = null, name = null } = originalError as any;
```

Comment:
Why not as Error?

Reply:
(also, could it be a string here or not?)

Reply:
If it's somehow a string, it will be caught at the end at made into the default 500 server error.

But I added defaults for when name is missing